# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieServiceParent.cpp
# Commit: 6379a4984eba
# Full Hash: 6379a4984ebafb659428ec3e4493f72173b3a70a
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-22 04:22:03
# Description:
#   Bug 1854028 - Update BounceTrackingStorageObserver cookie check and remove nsICookieNotification isThirdPartyCookie. r=bvandersloot,cookie-reviewers,anti-tracking-reviewers,timhuang
#   
#   For the BounceTrackingStorageObserver, instead of passing in a third-party flag
#   we can simply check if the changed cookie is partitioned or not.
#   
# ==============================================================================

diff -r 24c5f04d3793 -r 6379a4984eba netwerk/cookie/CookieServiceParent.cpp
--- a/netwerk/cookie/CookieServiceParent.cpp	Thu Sep 21 14:30:17 2023 +0000
+++ b/netwerk/cookie/CookieServiceParent.cpp	Thu Sep 21 14:42:32 2023 +0000
@@ -232,14 +232,13 @@
 IPCResult CookieServiceParent::RecvSetCookies(
     const nsCString& aBaseDomain, const OriginAttributes& aOriginAttributes,
     nsIURI* aHost, bool aFromHttp, const nsTArray<CookieStruct>& aCookies) {
-  return SetCookies(aBaseDomain, aOriginAttributes, aHost, aFromHttp, aCookies,
-                    0, false);
+  return SetCookies(aBaseDomain, aOriginAttributes, aHost, aFromHttp, aCookies);
 }
 
 IPCResult CookieServiceParent::SetCookies(
     const nsCString& aBaseDomain, const OriginAttributes& aOriginAttributes,
     nsIURI* aHost, bool aFromHttp, const nsTArray<CookieStruct>& aCookies,
-    uint64_t aBrowsingContextId, bool aIsThirdPartyCookie) {
+    dom::BrowsingContext* aBrowsingContext) {
   if (!mCookieService) {
     return IPC_OK();
   }
@@ -254,9 +253,9 @@
   // we don't send it back to the same content process.
   mProcessingCookie = true;
 
-  bool ok = mCookieService->SetCookiesFromIPC(
-      aBaseDomain, aOriginAttributes, aHost, aFromHttp, aCookies,
-      aBrowsingContextId, aIsThirdPartyCookie);
+  bool ok =
+      mCookieService->SetCookiesFromIPC(aBaseDomain, aOriginAttributes, aHost,
+                                        aFromHttp, aCookies, aBrowsingContext);
   mProcessingCookie = false;
   return ok ? IPC_OK() : IPC_FAIL(this, "Invalid cookie received.");
 }